src/main/java/gift/Controller/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package gift.Controller;
+
+import gift.model.Product;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/products")
+
+public class ProductController {
+
+  private final Map<Long, Product> products = new HashMap<>();
+  private Long id = 0L;
+
+  // 전체 상품 조회
+  @GetMapping
+  public List<Product> getAllProducts() {
+    return new ArrayList<>(products.values());
+  }
+
+  // 단건 상품 조회
+  @GetMapping("/{id}")
+  public ResponseEntity<Product> getProduct(@PathVariable Long id) {
+    Product product = products.get(id);
+    if (product == null) {
+      return ResponseEntity.notFound().build();
+    }
+    return ResponseEntity.ok(product);
+  }
+
+  // 상품 추가
+  @PostMapping
+  public ResponseEntity<Product> addProduct(@RequestBody Product product) {
+    product.setId(id);
+    products.put(id, product);
+    id++;
+    return ResponseEntity.ok(product);
+  }
```

Comment:
`ResponseEntity`를 사용해서 HTTP 상태 코드를 명시적으로 제어하는 부분은 아주 좋습니다! 👍
여기서 상품을 생성하는 addProduct 메소드를 조금 더 `RESTful`하게 만들어 볼 수 있을것 같아요 🤔

리소스가 성공적으로 생성되었을 때는 `201 Created` 상태 코드를 반환하는 것이 표준적인 REST API 설계인데요!
이때, 응답 헤더(Header)의 Location 필드에 새로 생성된 리소스에 접근할 수 있는 URI를 담아주면 클라이언트가 다음 동작을 이어가기 편리해지기에 추가해보는건 어떨까요? 😄

```suggestion
        // 현재 요청 URI를 기반으로 생성된 리소스의 URI를 동적으로 생성
        URI location = ServletUriComponentsBuilder.fromCurrentRequest()
            .path("/{id}")
            .buildAndExpand(id)
            .toUri();

    return ResponseEntity.created(location).body(product);
```